### ISSUE TITLE
Speed up interactive search

### ISSUE THREAD

**Issue by woiza** [Radarr/Radarr#10510]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Interactive search tends to be slow when there are many results. Additionally, only one CPU core is utilized, which I suspect is due to the sorting of the results.

### Describe the solution you'd like

Could this be improved by utilizing multiple CPU cores, for example, by using PLINQ?

### Describe alternatives you've considered

Use less indexers.

### Anything else?

[PLINQ](https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/introduction-to-plinq)

**Comment by bakerboy448** [comment]
- Define `slow`
- Trace logs proving where the `slowness` is?
- Most likely this is due to your indexers taking an extended amount of time to respond, trace logs would confirm - Radarr has no control over that

**Comment by woiza** [comment]
Thank you for your quick reply and suggestions.

I use three indexers, and their average response times range between 600 and 800 ms. My browser is Chrome (latest version), and my CPU is a Ryzen 5800X with 32 GB RAM. I run Radarr on a Rock 5B SBC (ARM, 8 cores) with 16 GB RAM and an NVMe SSD.

Below is a portion of my debug log. As you can see, there is a significant time difference in processing 63 reports versus 561 reports:

```
2024-10-03 19:06:31.1|Debug|ReleaseSearchService|Total of 63 reports were found for [REDACTED] from 3 indexers
2024-10-03 19:06:31.1|Debug|ReleaseSearchService|Setting last search time to: 10/03/2024 17:06:31
2024-10-03 19:06:31.1|Info|DownloadDecisionMaker|Processing 63 releases
...
2024-10-03 19:06:33.3|Debug|DownloadDecisionMaker|Release 'REDACTED' rejected for the following reasons: REDACTED
2024-10-03 19:06:33.3|Debug|Api|[GET] /api/v3/release?movieId=434: 200.OK (3115 ms)

...

2024-10-03 19:10:55.0|Debug|ReleaseSearchService|Total of 561 reports were found for [REDACTED] from 3 indexers
2024-10-03 19:10:55.0|Debug|ReleaseSearchService|Setting last search time to: 10/03/2024 17:10:55
2024-10-03 19:10:55.0|Info|DownloadDecisionMaker|Processing 561 releases
...
2024-10-03 19:11:16.2|Debug|DownloadDecisionMaker|Release 'REDACTED' rejected for the following reasons: REDACTED
2024-10-03 19:11:16.2|Debug|Api|[GET] /api/v3/release?movieId=426: 200.OK (25870 ms)
```

**Comment by woiza** [comment]
No, not at all! I am glad you took some time to look into this. I thought there might be some room for improvement, especially for people using low powered/clocked devices such as ARM SBCs, Intel N100, etc.

2 seconds for 63 releases and 13 seconds for 461 seems to be quite alright (my ARM CPU is only 2.4 GHz):

```
TMDB 937746

2024-10-03 20:37:10.1|Info|DownloadDecisionMaker|Processing 63 releases
2024-10-03 20:37:10.1|Trace|DownloadDecisionMaker|Processing release 1/63
...
2024-10-03 20:37:12.6|Trace|DownloadDecisionMaker|Processing release 63/63


TMDB 569094

2024-10-03 20:38:35.6|Info|DownloadDecisionMaker|Processing 461 releases
2024-10-03 20:38:35.6|Trace|DownloadDecisionMaker|Processing release 1/461
...
2024-10-03 20:38:48.9|Trace|DownloadDecisionMaker|Processing release 461/461
```

**Comment by woiza** [comment]
I see, than sorting is not the bottleneck but parsing and augmenting. Do you think this could be improved by splitting up the reports into several "buckets" and execute parsing and augmenting of these buckets on several CPU cores? Most SBCs and Intel NUCs have 4, 6, or even 8 cores...

**Comment by woiza** [comment]
Got it, thanks. Looking at htop, Radarr is single threaded (at least when I do an interactive search).
